### ISSUE TITLE
Need to migrate to Drive API before April 20, 2015

### ISSUE THREAD

**Issue by devotis** [jpillora/node-edit-google-spreadsheet#57]
Google recently communicated that Documents List API will be discontinued on April 20, 2015. This change means that service calls for this API will no longer be supported, and any Google Apps features that are implemented using this API will no longer function.

One of the URL's that will be discontinued is https://spreadsheets.google.com/feeds/ which is used by this module. It's suggested to migrate to the Drive API.

To summarize, node-edit-google-spreadsheet will no longer function as of April 20, 2015 if no action is taken.

**Comment by jpillora** [comment]
Are you sure? I had a look into this yesterday and made sure this module only uses Google Sheets API version 3.0 https://developers.google.com/google-apps/spreadsheets/ (no deprecation warning).

Which intuitively makes sense since the documents list API (the old google docs) is being replaced by Drive (the new google docs), though individual apps (documents, spreadsheets) remain the same.

https://spreadsheets.google.com/feeds/ an **authorization scope** used in Document List and that endpoint is never called. If you look down the doc list documentation, you'll see

> Retrieving all spreadsheets
> A list of only spreadsheets can be retrieved by using the spreadsheet category as follows:
> Protocol
> GET https://docs.google.com/feeds/default/private/full/-/spreadsheet

so as we can see here https://docs.google.com/feeds is the deprecated API

**Comment by devotis** [comment]
I drew my conclusion from line 144 in `lib/spreadsheet.js`
`url: this.protocol + '://spreadsheets.google.com/feeds/' +`

And this email that I received from Google recently
```
Important: Steps to migrate from discontinued Documents List API
Hello administrators,

We recently [communicated](http://googleappsupdates.blogspot.com/2014/12/reminder-to-migrate-to-updated-google.html)
that Documents List API will be discontinued on April 20, 2015.
This change means that service calls for this API will no longer be supported,
and any Google Apps features that are implemented using this API will no longer function.

Our records indicate that you may have an application that uses Documents List API,
and we recommend that you migrate to Drive API,
which has [comparable functionality](https://docs.google.com/a/google.com/spreadsheets/d/1j-eDNShUksN81dfpUjHLsYs3mYKzm-NcRETa28PqvTQ/pubhtml#),
as soon as possible.

Here's what you need to do:

Determine if you have an application that makes requests to these types of URLs:

https://docs.google.com/feeds/
https://docs.googleusercontent.com/
https://spreadsheets.google.com/feeds/

Migrate the applications to [Drive API](https://developers.google.com/drive/).
If you have questions about migration, please contact Google Apps for Work Support.

Sincerely,
The Google Apps for Work Team
```

**Comment by jpillora** [comment]
That's an interesting email. I'm thinking it must be a typo... Since [this](https://github.com/jpillora/node-edit-google-spreadsheet/blob/stable/lib/spreadsheet.js#L200) is the base URL for all endpoints which read and write spreadsheets and there is no read and write spreadsheet API in Drive. Drive seems to only have user access, files and folders related APIs. So I don't understand how one could "move" to the Drive API? Can you reply to the email and ask?

**Comment by devotis** [comment]
I opened a case at enterprise.google.com.
Case 05364216 - Spreadsheet URLs mentioned in email about discontinued Documents List API

You're on cc (dev at ...)

**Comment by jpillora** [comment]
Just read it, thanks!

**Comment by devotis** [comment]
Maybe I just read it the other way around and it says that you're OK when you're using the URL's mentioned. Anyway I posted question to here: 
http://stackoverflow.com/questions/28169309/do-you-need-to-migrate-to-drive-api-when-using-https-spreadsheets-google-com-f

**Comment by jpillora** [comment]
Cool looks good, hopefully some one from Google responds

**Comment by jpillora** [comment]
Thanks for checking up on this. From google support: "If you're only using the spreadsheet scope, you don't have to change your application"

**Comment by devotis** [comment]
Thank you for your support.
